src/axom/core/memory_management.hpp

```diff
@@ -198,7 +198,7 @@ inline T* reallocate( T* pointer, std::size_t n ) noexcept
 {
   const std::size_t numbytes = n * sizeof( T );
 
-#ifdef AXOM_USE_UMPIRE
+#if defined(AXOM_USE_UMPIRE) && !defined(UMPIRE_VERSION_MAJOR)
```

Comment:
Please add a comment here that ``UMPIRE_VERSION_MAJOR`` was added in Umpire v2.0.0, so if it is not defined, then we're assuming that we need the workaround.

Reply:
Thanks, I added a comment to clarify that.